### PR TITLE
Switch to the new `tmt` logo

### PR DESCRIPTION
**Pull request by martinhoyer**
Dependent on https://github.com/teemtee/docs/pull/9 See for additional context.
Cleaned up some (I presume) obsolete logo-related-steps from conf.py

Pull Request Checklist

* [x] implement the feature
* [x] include a release note

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -26,12 +26,6 @@ STEPS = discover provision prepare execute finish report
 # A list of `plugins/*.rst` files to generate
 PLUGIN_TARGETS = $(addsuffix .rst,$(addprefix plugins/,$(STEPS))) plugins/test-checks.rst
 
-# A source of logo picture for Sphinx docs favicon
-LOGO_SRC = https://raw.githubusercontent.com/teemtee/docs/main/logo/tmt-small.png
-
-# Local filepath to fetched logo
-LOGO_DST = _static/tmt-small.png
-
```
Comment (psss) [inline docs/Makefile:34]: Seems ok to drop the logo download. Unless github refuses to serve the raw content to the regular users as well ;-) @happz, any concerns here?

**Review by psss — COMMENTED** [review]
Thanks, looks good, two questions though.

File: docs/Makefile (hunk 1 above)
Comment (happz) [reply docs/Makefile:34]: I believe the downloaded file was not used at all. It was replaced with the remote one in https://github.com/teemtee/tmt/commit/9d19728b084d184e5727ab443d222402f46e7bd9

File: docs/conf.py [hunk 2]
```diff
@@ -217,12 +217,11 @@ def _load_theme(
...
 # of the sidebar.
-html_logo = 'https://raw.githubusercontent.com/teemtee/docs/main/logo/tmt-transparent-175x175.png'
+html_logo = 'https://raw.githubusercontent.com/teemtee/docs/main/logo/tmt-logo-dark-background.png'
 
 # The name of an image file (within the static path) to use as favicon of the
-# docs.  This file should be a Windows icon file (.ico) being 16x16 or 32x32
-# pixels large.
-html_favicon = 'https://raw.githubusercontent.com/teemtee/docs/main/logo/tmt-favicon.ico'
+# docs.
+html_favicon = 'https://raw.githubusercontent.com/teemtee/docs/main/logo/tmt-logo-dark-background.svg'
```
[the remark this answers is not in the document]
Comment (psss) [reply docs/conf.py:224]: And I've found the answer on the chat:
> ha, it looks like all browsers now support .svg favicons. Much better than .ico

File: docs/Makefile (hunk 1 above)
Comment (psss) [reply docs/Makefile:34]: Ack, and if github decides to stop serving the raw files, we'll fix it. Hope they will not.

**Pull request merged**